src/encoded/static/components/cart/cart.js

```diff
@@ -156,27 +184,27 @@ CartSearchResults.defaultProps = {
  * Display browser tracks for the selected page of files.
  */
 const CartBrowser = ({ files, assembly, pageNumber }) => {
-    const [pageFiles, setPageFiles] = React.useState([]);
-
-    // Request full file objects for the genome browser but only for the given page of file @ids.
-    React.useEffect(() => {
-        if (files.length > 0 && assembly) {
-            const pageStartingIndex = pageNumber * PAGE_TRACK_COUNT;
-            const pageFilesIds = files.slice(pageStartingIndex, pageStartingIndex + PAGE_TRACK_COUNT);
-            requestObjects(pageFilesIds, '/search/?type=File').then((fileResults) => {
-                setPageFiles(fileResults);
-            });
-        } else {
-            // Clear genome browser if new props have no visualizable tracks.
-            setPageFiles([]);
+    // Extract the current page of file objects.
+    const pageStartingIndex = pageNumber * PAGE_TRACK_COUNT;
+    const pageFiles = files.slice(pageStartingIndex, pageStartingIndex + PAGE_TRACK_COUNT);
+
+    // Shorten long annotation_type values for the Valis track label; fine to mutate `pageFiles` as
+    // it holds a copy of a segment of `files`.
+    pageFiles.forEach((file) => {
+        if (file.annotation_type) {
+            const mappedAnnotationType = annotationTypeMap[file.annotation_type];
+            if (mappedAnnotationType) {
+                file.annotation_type = mappedAnnotationType;
+            }
         }
-    }, [files, assembly, pageNumber]);
+    });
+
     const sortParam = ['Assay term name', 'Biosample term name', 'Output type'];
     return <GenomeBrowser files={pageFiles} label={'cart'} assembly={assembly} expanded sortParam={sortParam} />;
 };
 
 CartBrowser.propTypes = {
-    /** File @ids of all visualizable tracks, not just on the displayed page */
+    /** Files of all visualizable tracks, not just on the displayed page */
```

Comment:
Did we want to add the sort buttons now, since we're collecting all the visualizable tracks?